### DISCUSSION TITLE
contentTypes multiple identical field groups

### DISCUSSION THREAD

**Discussion by cereal2nd** [bolt/core#2401] [Q&A] [answered]
Hi,

i'm trying to create some sort of 'event' content type, this event will consist out of a multiple groups
- main part (name, url, date, .... fields)
- artist 1
- artist 2
- artist 3

now the fields dor these artists are exactly the same, so do i have to duplicate all these fields? or is there a way to reuse a group of fields multiple times in the same content type?

**Comment by I-Valchev** [comment, marked as answer]
Hey @Cereal2nd , welcome to Bolt! There are a couple of things to consider here.

1. Are the artists the same across different events? If yes, you may want to create an Artist content type, which is configured with `viewless: true` and you can "link" each artist to an event using Relations or a Select field
2. If the artists are not the same (or you want them completely separate for another reason), then you're most likely looking for a [Collection field](https://docs.bolt.cm/4.0/fields/collection#jumpbutton). The collection will repeat, as many times as you wish, the fields that you define in it. It's often used together with a [Set field](https://docs.bolt.cm/4.0/fields/set), which groups several other fields into one.

For example, a collection of any number of Authors, where each Author has a name and a date of birth can look like this:

```yaml
authors:
    type: collection
    fields:
        author:
        type: set
        fields:
            name:
                type: text
            dob:
                type: date
```

In short, a Collection gives you the _repeat_ functionality, a Set gives you the _group_ functionality. Hope this helps!

**Reply by cereal2nd** [reply]
thanks a lot, this was verry usefull and exactly what i needed

**Reply by I-Valchev** [reply]
Awesome, glad to hear! :-)
